### ISSUE TITLE
[VPAT][A11y] datepicker current selected year is missing state announcement

### ISSUE THREAD

**Issue by chunhtai** [flutter/flutter#173055]
b/429062825

Run the https://github.com/flutter/flutter/tree/master/dev/a11y_assessments and navigate to datepicker

<img width="1017" height="1800" alt="Image" src="https://github.com/user-attachments/assets/b0fee56b-04aa-4aa9-aac5-c307111b6a96" />

EXPECTED RESULTS:

When focus on current selected year it should announce that it is currently selected

VoiceOver announces "2025, Current Year, Button".

**Comment by huycozy** [comment]
Tested this on the latest Flutter stable 3.35.3, with Android TalkBack reader, I heard it announce the selected state as expected: _"selected, 2025 button, double tap to activate"_. On iOS (I tested with XCode's Accessibility Inspector), it announces the selected state too, only the order is changed: "2025 button, selected".

<details>
<summary>Demo screenshot/video</summary>

| Android | iOS |
| --------------- | --------------- |
<video src="https://github.com/user-attachments/assets/5b9d8574-432c-43f8-824d-e91a1edca7c4"/> | <video src="https://github.com/user-attachments/assets/80d45bd4-5764-4179-b241-c789d70a6ceb"/>

</details>


I also looked into the code and saw the `Semantics.selected` is set as well:

https://github.com/flutter/flutter/blob/a04fb324be734cf18811c30c06baf9bf07b3bab3/packages/flutter/lib/src/material/calendar_date_picker.dart#L1483

Austin has implemented this since 2020 in https://github.com/flutter/flutter/pull/50546, so it's not a fix that landed recently. Can you retry, @chunhtai?
